cfme/fixtures/v2v.py

```diff
@@ -370,119 +328,106 @@ def form_data_vm_obj_dual_nics(request, appliance, source_provider, provider):
-            },
-            "datastore": {
-                "Cluster ({})".format(provider.data.get("clusters")[0]): {
-                    "mappings": [
-                        _form_data_mapping(
-                            "datastores",
-                            source_provider,
-                            provider,
-                            request.param[0],
-                            request.param[1],
-                        )
-                    ]
-                }
-            },
+            "description": "Single DS migration of VM from {ds_type1} to {ds_type2},".format(
+                ds_type1=request.param[0], ds_type2=request.param[1]
+            ),
+            "datastores": [
+                component_generator(
+                    "datastores", source_provider, provider, request.param[0], request.param[1]
+                )
+            ],
         },
     )
     vm_obj = get_vm(request, appliance, source_provider, request.param[2], request.param[0])
-    return FormDataVmObj(form_data=form_data, vm_list=[vm_obj])
+    return FormDataVmObj(infra_mapping_data=infra_mapping_data, vm_list=[vm_obj])
 
 
 @pytest.fixture(scope="function")
 def form_data_vm_obj_single_network(request, appliance, source_provider, provider):
-    form_data = _form_data(source_provider, provider)
+    infra_mapping_data = infra_mapping_default_data(source_provider, provider)
     recursive_update(
-        form_data,
+        infra_mapping_data,
         {
-            "general": {
-                "description": "Single Network migration of VM from {vlan1} to {vlan2},".format(
-                    vlan1=request.param[0], vlan2=request.param[1]
+            "description": "Single Network migration of VM from {vlan1} to {vlan2},".format(
+                vlan1=request.param[0], vlan2=request.param[1]
+            ),
+            "networks": [
+                component_generator(
+                    "vlans", source_provider, provider, request.param[0], request.param[1]
                 )
-            },
-            "network": {
-                "Cluster ({})".format(provider.data.get("clusters")[0]): {
-                    "mappings": [
-                        _form_data_mapping(
-                            "vlans", source_provider, provider, request.param[0], request.param[1]
-                        )
-                    ]
-                }
-            },
+            ],
         },
     )
     vm_obj = get_vm(request, appliance, source_provider, request.param[2])
-    return FormDataVmObj(form_data=form_data, vm_list=[vm_obj])
+    return FormDataVmObj(infra_mapping_data=infra_mapping_data, vm_list=[vm_obj])
 
 
-def _form_data_mapping(selector, source_provider, provider, source_list=None, target_list=None):
+def component_generator(selector, source_provider, provider, source_list=None, target_list=None):
+    """
+    Component generator method to generate a dict of source and target
+    components(cluster/datastore/networks).
```

Comment:
Please expand this description a bit as the function is doing a few things:

Its looking up component names in provider yaml, and skipping the test is relevant information is not provided.